src/crystal/dwarf/line_numbers.cr

```diff
@@ -240,6 +282,100 @@ module Crystal
         end
       end
 
+      record LNCTFormat,
+        lnct : LNCT,
+        format : FORM
+
+      enum LNCT : UInt32
+        PATH            = 0x01
+        DIRECTORY_INDEX = 0x02
+        TIMESTAMP       = 0x03
+        SIZE            = 0x04
+        MD5             = 0x05
+      end
+
+      private def read_lnct_format
+        count = @io.read_bytes(UInt8)
+        Array(LNCTFormat).new(count) do
+          LNCTFormat.new(
+            lnct: LNCT.new(DWARF.read_unsigned_leb128(@io)),
+            format: FORM.new(DWARF.read_unsigned_leb128(@io))
+          )
+        end
+      end
+
+      private def read_lnct(sequence, formats)
+        dir = ""
+        path : String = ""
+        mtime : UInt64 = 0_u64
+        size : UInt64 = 0_u64
```

Comment:
Why specify types when we already have _explicit_ default? It stands out since it's used nowhere else in the file...

Reply:
The type declarations are there to make sure that no other types can sneak in and create a union. This might happen when adding support for more formats.
I suppose, adding return type restrictions should have the same effect.
Or maybe we should introduce a record type which can also be used in the `file_names` list, instead of a tuple.

Reply:
I think the latter would really be a great, because I already messed up the tuple format.